File: Algorithm-CPP/40_교집합.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n;
    cin >> n;

    int arr1[n];
    for(int i = 0; i < n; i++) cin >> arr1[i];
    for(int i = 1; i < n; i++) {
        for(int j = i; j > 0; j--) {
            if(arr1[j - 1] < arr1[j]) break;

            int temp = arr1[j];
            arr1[j] = arr1[j - 1];
            arr1[j - 1] = temp;
        }
    }

    int m;
    cin >> m;

    int arr2[m];
    for(int i = 0; i < m; i++) cin >> arr2[i];
    for(int i = 1; i < m; i++) {
        for(int j = i; j > 0; j--) {
            if(arr2[j - 1] < arr2[j]) break;

            int temp = arr2[j];
            arr2[j] = arr2[j - 1];
            arr2[j - 1] = temp;
        }
    }

    int i = 0, j = 0;
    vector<int> res;
    while(i != n && j != m) {
        if(arr1[i] == arr2[j]) {
            res.push_back(arr1[i]);
            i++; j++;
        } 
        else if(arr1[i] > arr2[j]) j++;
        else if(arr1[i] < arr2[j]) i++;
    }

    for(int i = 0; i < res.size(); i++) cout << res[i] << " ";
    
    return 0;
}

File: Algorithm-Swift/Algorithm-Swift/Inflearn/40_교집합.swift
//
//  40_교집합.swift
//  Algorithm-Swift
//
//  Created by SHIN YOON AH on 2023/08/17.
//

import Foundation

func solution40() {
    let n = Int(readLine()!)!
    let arr1 = readLine()!.split(separator: " ").map { Int($0)! }.sorted()
    let m = Int(readLine()!)!
    let arr2 = readLine()!.split(separator: " ").map { Int($0)! }.sorted()
    var i = 0, j = 0, res = ""

    while i != n && j != m {
        if arr1[i] == arr2[j] {
            res += "\(arr1[i]) "
            i += 1
            j += 1
        }
        else if arr1[i] > arr2[j] { j += 1 }
        else if arr1[i] < arr2[j] { i += 1 }
    }

    print(res)
}
